src/call.cpp

```diff
@@ -985,4 +985,46 @@ void DarlingServer::Call::S2CPerform::processCall() {
 	_sendReply(code);
 };
 
+void DarlingServer::Call::SetExecutablePath::processCall() {
+	int code = 0;
+	int fullLength = 0;
+
+	if (auto thread = _thread.lock()) {
+		if (auto process = thread->process()) {
+			std::string tmpstr;
+			tmpstr.resize(_body.buffer_size);
+			if (!process->readMemory((uintptr_t)_body.buffer, tmpstr.data(), _body.buffer_size, &code)) {
+				code = -code;
+			}
+			process->setExecutablePath(tmpstr.c_str());
+		} else {
+			code = -ESRCH;
+		}
+	} else {
+		code = -ESRCH;
+	}
+
+	_sendReply(code == 0 ? fullLength : code);
```

Comment:
The reply code is only meant to return error codes or `0` for success; besides, it looks like you're never modifying `fullLength`, so you're just returning `0` anyways.
```suggestion
	_sendReply(code);
```